mono/metadata/marshal.c

```diff
@@ -7594,8 +7594,10 @@ typedef enum {
 	/* Initialized an object handle to null, pass to the icalls,
 	   write the value out from the handle when the icall returns */
 	ICALL_HANDLES_WRAP_OBJ_OUT,
-	/* Wrap the argument (a valuetype reference) in a handle to pin its enclosing object,
-	   but pass the raw reference to the icall */
+	/* Wrap the argument (a valuetype reference) in a handle to pin its
+	   enclosing object, but pass the raw reference to the icall.  This is
+	   also how we pass byref generic parameter arguments to generic method
+	   icalls (eg, System.Array:GetGenericValueImpl<T>(int idx, T out value)) */
 	ICALL_HANDLES_WRAP_VALUETYPE_REF,
```

Comment:
Where is this used ?

Reply:
It will allow us to merge https://github.com/mono/mono/pull/6907